src/Chainweb/Pact/SPV/Hyperlane/Binary.hs

```diff
@@ -0,0 +1,190 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Chainweb.Pact.SPV.Hyperlane.Binary where
+
+import Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+
+import Data.DoubleWord
+import Data.Text (Text)
+import qualified Data.Text.Encoding as Text
+import Data.Binary
+import qualified Data.Binary.Builder as Builder
+import Data.Binary.Get
+import Data.Binary.Put
+
+data HyperlaneMessage = HyperlaneMessage
+  { hmVersion :: Word8            -- uint8
+  , hmNonce :: Word32             -- uint32
+  , hmOriginDomain :: Word32      -- uint32
+  , hmSender :: BS.ByteString     -- bytes32
+  , hmDestinationDomain :: Word32 -- uint32
+  , hmRecipient :: BS.ByteString  -- bytes32
+  , hmTokenMessage :: TokenMessageERC20
+  }
+
+-- Corresponds to abi.encodePacked behaviour
+instance Binary HyperlaneMessage where
```

Comment:
Please replace these binary instances with functions, and use Chainweb.Utils.Serialization instead of Data.Binary.